### PR TITLE
Feature/prisma upgrade

### PR DESCRIPTION
**Pull request by SOULPWRD**
This PR includes:

- Upgrade all prisma related packages to v4
- Upgrade most of the package to the latest versions
- Replace `@prisma/sdk` with `@prisma/internals`
- Keep `typescript` version due to the jest error

### CODE REVIEW

File: src/test/functional/basic/expected/spreadsheet.ts [hunk 1]
```diff
@@ -1,16 +1,10 @@
...
-type Json = Literal | { [key: string]: Json } | Json[]
-const literalSchema = z.union([z.string(), z.number(), z.boolean()])
-const jsonSchema: z.ZodSchema<Json> = z.lazy(() => z.union([literalSchema, z.array(jsonSchema), z.record(jsonSchema)]))
-
 export const SpreadsheetModel = z.object({
   id: z.string(),
   filename: z.string(),
   author: z.string(),
-  contents: jsonSchema,
+  contents: z.any(),
```
Comment (abenhamdine) [inline src/test/functional/basic/expected/spreadsheet.ts:7]: Why this change ?
Also it seems unrelated to prisma 4 migration.

File: src/test/functional/imports/expected/document.ts [hunk 2]
```diff
@@ -1,12 +1,12 @@
 import * as z from "zod"
-import { Status } from "../prisma/.client"
+import prisma from "../prisma/.client"
```
Comment (abenhamdine) [inline src/test/functional/imports/expected/document.ts:2]: Why this change ?

File: src/test/regressions.test.ts [hunk 3]
```diff
@@ -35,7 +35,7 @@ describe('Regression Tests', () => {
 		writeImportsForModel(model, testFile, config, prismaOptions)
 
 		expect(testFile.print()).toBe(
-			'import * as z from "zod";\nimport { UserType } from "@prisma/client";\n'
+			'import * as z from "zod";\nimport prisma from "@prisma/client";\n'
```
Comment (abenhamdine) [inline src/test/regressions.test.ts:38]: why this change ?

File: src/types.ts [hunk 4]
```diff
@@ -41,7 +41,7 @@ export const getZodConstructor = (
 				break
 		}
 	} else if (field.kind === 'enum') {
-		zodType = `z.nativeEnum(${field.type})`
+		zodType = `z.nativeEnum(prisma.${field.type})`
```
Comment (abenhamdine) [inline src/types.ts:44]: why this change ?

File: tsconfig.json [hunk 5]
```diff
@@ -28,6 +28,7 @@
 		"noEmit": true,
 		"downlevelIteration": true,
 		"resolveJsonModule": true,
+		"allowSyntheticDefaultImports": true,
```
Comment (abenhamdine) [inline tsconfig.json:31]: why should we need this change ?

**Comment by SOULPWRD** [comment]
@abenhamdine Sorry this was an unintentional change. This change was supposed to be for our internal fork of this repo. My bad. Closing this PR.
